content-src/lib/selectLayoutRender.js

```diff
@@ -45,7 +45,16 @@ export const selectLayoutRender = createSelector(
           return component;
         }
 
-        return {...component, data: maybeInjectSpocs(feeds.data[component.feed.url].data, component.spocs)};
+        let {data} = feeds.data[component.feed.url];
+
+        if (component && component.properties && component.properties.offset) {
+          data = {
+            ...data,
+            recommendations: data.recommendations.slice(component.properties.offset),
+          };
+        }
+
+        return {...component, data: maybeInjectSpocs(data, component.spocs)};
```

Comment:
These changes are further optimizations to lower the number of feed requests.

If the server wants to combine feeds, they can now. Because I've added:

1. Duplicate feeds with the same url don't get re fetched.
2. If two components use the same feed, they can be told to use a different segment from the same data.

It's still on the server to ensure the offset param doesn't allow dupes. So we can essentially combine smaller feeds into one, the components can use the same url, and the data would be the same, so the offset would cut out unique slices from the data.

Reply:
Truly hawt -- nice work!